src/commonMain/kotlin/fr/acinq/lightning/channel/Helpers.kt

```diff
@@ -246,6 +244,38 @@ object Helpers {
         }
     }
 
+    /**
+     * Return the list of wallet inputs used in pending unconfirmed channel funding attempts.
+     * These inputs should not be reused in other funding attempts, otherwise we would double-spend ourselves.
+     */
+    fun reservedWalletInputs(channels: List<PersistedChannelState>): Set<OutPoint> {
+        return channels.filterNot { it is Closed }.flatMap { channel ->
+            when (channel) {
+                is WaitForFundingSigned -> channel.signingSession.fundingTx.tx.localInputs.map { it.outPoint }
+                is ChannelStateWithCommitments -> {
+                    val unsignedInputs = when (channel) {
+                        is WaitForFundingConfirmed -> when (val rbfStatus = channel.rbfStatus) {
+                            is RbfStatus.WaitingForSigs -> rbfStatus.session.fundingTx.tx.localInputs.map { it.outPoint }
+                            else -> listOf()
+                        }
+                        is Normal -> when (val spliceStatus = channel.spliceStatus) {
+                            is SpliceStatus.WaitingForSigs -> spliceStatus.session.fundingTx.tx.localInputs.map { it.outPoint }
+                            else -> listOf()
+                        }
+                        else -> listOf()
+                    }
+                    val fundingInputs = channel.commitments.all.flatMap { commitment ->
+                        when (val fundingStatus = commitment.localFundingStatus) {
+                            is LocalFundingStatus.UnconfirmedFundingTx -> fundingStatus.sharedTx.tx.localInputs.map { it.outPoint }
+                            is LocalFundingStatus.ConfirmedFundingTx -> listOf() // utxos spent by confirmed transactions will never appear in our wallet
+                        }
+                    }
+                    unsignedInputs + fundingInputs
+                }
+            }
+        }.toSet()
```

Comment:
Same behavior, written in two separate stages instead of imbricated. I think it conveys the intent better.

```suggestion
        val fundingTxs: List<SignedSharedTransaction> = buildList {
            for (channel in channels) {
                // add all unsigned inputs currently used to build a funding tx (creation, rbf, splice)
                when {
                    channel is WaitForFundingSigned -> add(channel.signingSession.fundingTx)
                    channel is WaitForFundingConfirmed && channel.rbfStatus is RbfStatus.WaitingForSigs -> channel.rbfStatus.session.fundingTx
                    channel is Normal && channel.spliceStatus is SpliceStatus.WaitingForSigs -> channel.spliceStatus.session.fundingTx
                    else -> {}
                }
                // add all inputs in unconfirmed funding txs (utxos spent by confirmed transactions will never appear in our wallet)
                when {
                    channel is ChannelStateWithCommitments -> channel.commitments.all
                        .map { it.localFundingStatus }
                        .filterIsInstance<LocalFundingStatus.UnconfirmedFundingTx>()
                        .forEach { add(it.sharedTx) }
                    else -> {}
                }
            }
        }
        val localInputs = fundingTxs.flatMap { fundingTx -> fundingTx.tx.localInputs.map { it.outPoint } }
        return localInputs.toSet()
```

Reply:
This is better, thanks! Done in https://github.com/ACINQ/lightning-kmp/pull/474/commits/88f3c72e4bcc145faccfcb236df7bc3b4512e475